### PR TITLE
issue: 4197110: Add the collectx versions to the to the PDF

### PR DESCRIPTION
**Pull request by Miryam-Schwartz**
## What
_Add the collectx versions to the to the PDF._ 

## Why ?
_Display of extracted data._

## How ?
_save_collectx_versions method was added to the class set _funcs_for_analysis. This method extracts the versions and save them in _dataframes_for_pdf list._

## Testing ?
_Tested manually to ensure the output looks the same as before the change._

## Special triggers
_Use the following phrases as comments to trigger different runs_

* ```bot:retest``` rerun Jenkins CI (to rerun GitHub CI, use "Checks" tab on PR page and rerun _all_ jobs)
* ```bot:upgrade``` run additional update tests

### CODE REVIEW

File: plugins/ufm_log_analyzer_plugin/src/loganalyze/log_analyzers/ibdiagnet2_port_counters_analyzer.py [hunk 1]
```diff
@@ -52,11 +53,16 @@ def __init__(
...
 
-    def get_collectx_versions(self):
+    def save_collectx_versions(self):
         unique_collectx_versions = self._log_data_sorted[
             self._log_data_sorted["type"] == "collectx_version"
         ]["data"].unique()
-        return unique_collectx_versions
+        self._dataframes_for_pdf.append(
+            (
+                f"{self.telemetry_type} telemetry collectx versions",
```
Comment (boazhaim) [inline plugins/ufm_log_analyzer_plugin/src/loganalyze/log_analyzers/ibdiagnet2_port_counters_analyzer.py:62]: collectx versions found in {self.telemetry_type} telemetry log

**Review by boazhaim — APPROVED** [review]

File: plugins/ufm_log_analyzer_plugin/src/loganalyze/pdf_creator.py [hunk 2]
```diff
@@ -60,7 +60,8 @@ def add_images(self):
     def add_text(self):
         self.set_font("Arial", "", 12)
         output = StringIO()
-        print(self._fabric_stats_list, file=output)
+        for txt in self._txt_list:
+            print(txt, file=output)
```
Comment (boazhaim) [inline plugins/ufm_log_analyzer_plugin/src/loganalyze/pdf_creator.py:64]: If you touched it :) (Which is great!)
Why do we need this logic? Cant we add these texts and combine them with the text we send in add_list_of_dicts_as_text ? 
(will also mean a small refactor, but it makes sense)
Not a must right now...

File: plugins/ufm_log_analyzer_plugin/src/loganalyze/log_analyzers/console_log_analyzer.py [hunk 3]
```diff
@@ -106,6 +109,11 @@ def print_exceptions_per_time_count(self):
             "Exceptions count",
         )
 
+    def save_ufm_versions(self):
+        self._txt_for_pdf.append(
+            f"Used ufm version in console log {self.ufm_versions}{os.linesep}"
```
Comment (boazhaim) [inline plugins/ufm_log_analyzer_plugin/src/loganalyze/log_analyzers/console_log_analyzer.py:114]: With the change you did, I'm not sure we need the os.linesep here. And if we do, it does not make sense to have it here, I would move it to the PDF class

File: plugins/ufm_log_analyzer_plugin/src/loganalyze/log_analyzers/ibdiagnet2_port_counters_analyzer.py (hunk 1 above)
Comment (Miryam-Schwartz) [reply plugins/ufm_log_analyzer_plugin/src/loganalyze/log_analyzers/ibdiagnet2_port_counters_analyzer.py]: fixed

**Pull request merged**